src/widgets/SearchBar.js

```diff
@@ -17,14 +17,17 @@ import { debounce } from '../utilities/index';
  * Simple search bar - essentially a wrapper around a text input
  * with a magnifying glass icon and clear button.
  *
- * @param {String} color        Color of the entire component (monochrome only).
- * @param {String} onChangeText Callback for changing text (debounced).
- * @param {String} value        Text value of the search input.
- * @param {String} placeholder  Placeholder value.
- * @param {String} autoFocus
- * @param {String} textInputStyle
- * @param {String} viewStyle
- * @param {String} debounceTimeout
+ * Debounces input by the user, such that the onChangeText callback
+ * is only invoked once every `debounceTimeout`
+ *
+ * @param {String} color           Color of the entire component (monochrome only).
+ * @param {String} onChangeText    Callback for changing text (debounced).
+ * @param {String} value           Text value of the search input.
+ * @param {String} placeholder     Placeholder value.
+ * @param {String} autoFocus       Autofocus this component on mounting.
+ * @param {String} textInputStyle  Style of the TextInput component.
+ * @param {String} viewStyle       Style of the wrapping View component.
+ * @param {String} debounceTimeout Time in milliseconds to debounce the onChangeText callback.
```

Comment:
Could be dumb, but maybe a HOC like:

```javascript
withDebounce = (component, timeout, [callbackPropsToDebounce]) => {...}
```

Not sure how HOCs like that might play with hooks.

Reply:
hm it's quite cool - maybe I'll have a play with something